core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -34,10 +44,34 @@ oppia.factory('UrlService', ['$window', function($window) {
       return urlParts[1] === 'embed';
     },
     getPathname: function() {
-      return $window.location.pathname;
+      return this.getCurrentLocation().pathname;
+    },
+    getQueryFieldValuesAsList: function(fieldName) {
+      var fieldValues = [];
+      if (this.getCurrentHref().indexOf('?') > -1) {
+        // Each queryItem return one field-value pair in the url.
+        var queryItems = this.getCurrentHref().slice(
+          this.getCurrentHref().indexOf('?') + 1).split('&');
+        for (var i = 0; i < queryItems.length; i++) {
+          var currentFieldName = decodeURIComponent(
+            queryItems[i].split('=')[0]);
+          var currentFieldValue = decodeURIComponent(
+            queryItems[i].split('=')[1]);
+          if (currentFieldName === fieldName) {
+            fieldValues.push(currentFieldValue);
+          }
+        }
+      }
+      return fieldValues;
+    },
+    addField: function(url, fieldName, fieldValue) {
+      fieldValue = encodeURIComponent(fieldValue);
```

Comment:
Don't redeclare; use a new varname like encodedFieldValue. This should be a general practice, otherwise you'd need to trace the code carefully to figure out the meaning of a variable at a particular point in the code.

Reply:
Ok, will remember that, done.